Dockerfile

```diff
@@ -1,24 +1,39 @@
-FROM golang:latest
-MAINTAINER Barak Michener <me@barakmich.com>
+FROM alpine:3.5
 
-# Set up workdir
-WORKDIR /go/src/github.com/cayleygraph/cayley
+LABEL maintainer "napalmbrain (psev)"
+LABEL email "github@napalmbrain.sugarush.io"
+LABEL github "github.com/sugarush"
+LABEL twitter "@iamnapalmbrain"
 
-# Restore vendored dependencies
-RUN sh -c "curl https://glide.sh/get | sh"
-ADD glide.* ./
-RUN glide install
+ENV INSTALL_PATH="/usr/local/bin"
 
-# Add and install cayley
-ADD . .
-RUN go install -v ./cmd/cayley
+ENV PKG="ca-certificates" \
+    PKG_TMP="g++ git go glide" \
+    PKG_CACHE="/var/cache/apk"
 
-# Expose the port and volume for configuration and data persistence. If you're
-# using a backend like bolt, make sure the file is saved to this directory.
-RUN mkdir /data #VOLUME ["/data"]
-EXPOSE 64210
+ENV BUILD_LIB="github.com/cayleygraph/cayley" \
+    BUILD_TAG="master"
```

Comment:
Docker builds should be decoupled from the source code. This allows that exactly.